### PR TITLE
fixed doc generated on rtd

### PR DESCRIPTION
**Pull request by shrutij01**
Some fixes in the doc files, a yml for importing the conda env into rtd, and spefication of the package root.
Generated docs look like [this](https://trifinger-robot-simulation.readthedocs.io/en/latest/index.html).

## How I Tested

By building on rtd

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -9,55 +9,14 @@ There are two ways to build/install this package.
...
-
-    Every time you activate your trifinger_simulation conda environment, it is
-    important that you make sure that ``PYTHONNOUSERSITE`` is set to True, as is
-    done in step 2 above.  This ensures that your conda environment is isolated
-    from the global/user site-packages, which is important for us to reproduce
-    your env on our side. For convenience, we provide a script which does this::
-
-        source conda_activate_trifinger_simulation.sh
-
+Please refer to the `repository readme`_ for instructions.
```
Comment (luator) [inline docs/installation.rst:12]: I would rather do it the other way round, i.e. point from the readme to the docs.  I think the documentation should be self-contained and not link to other resources for something essential like installation instructions.

File: docs/timesteplogic.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+
+In the software, the method
+:meth:`~trifinger_simulation.SimFinger.append_desired_action` is used to
+append actions to the time series.  It returns the time index *t* at which the
+appended action will be executed.  Methods like
+:meth:`~trifinger_simulation.SimFinger.get_observation` expect a time index as
+input and will return the data corresponding to this time step.  If the given
+time index refers to a point in the future, these methods will block and wait
+until that point is reached.
```
Comment (luator) [inline docs/timesteplogic.rst:27]: This is not true for the simulation, only for the robot_interfaces counterpart.

File: docs/timesteplogic.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  follows the same principle to make the transition to the real robot easier.
+  However, it is implemented with a buffer size of 1, so the getter methods only
+  provide data for the current time step.
+- It is possible to access information from *t + 1*.  In a typical gym
+  environment, it is expected that the observation returned by ``step(action)``
+  belongs to the moment *after* the given action is executed (this corresponds
+  to the time index *t + 1*).  To make it easier to get started, we therefore
+  allow to access the observations of this time index in the simulation.
+
+Desired vs Applied Action
```
Comment (luator) [inline docs/timesteplogic.rst:67]: This section is not really related to time step logic.  It would probably make sense to move it to the file about the `SimFinger` class.

File: python/trifinger_simulation/sim_finger.py [hunk 4]
```diff
@@ -18,6 +18,16 @@ class SimFinger:
...
+
+    Attributes:
+        finger_type (string): Name of the finger type.  Use
+                :meth:`get_valid_finger_types` to get a list of all supported
+                types.
+        time_step (float): Time (in seconds) between two simulation steps.
+            Don't set this to be larger than 1/60.  The gains etc. are set
+            according to a time_step of 0.004 s.
+        enable_visualization (bool): Set this to 'True' for a GUI interface
+            to the simulation.
```
Comment (luator) [inline python/trifinger_simulation/sim_finger.py:30]: I would rather add this in the code below, using `#:` above the atrributes.  Then there is less distance between code and documentation which makes it less likely to get out of sync.

Comment (shrutij01) [reply python/trifinger_simulation/sim_finger.py:30]: Ok, I added it like this to be similar to the way it is in Action and Observation. Can we keep it like this for now, and then change all of these consistently later?

File: docs/installation.rst [hunk 5]
```diff
@@ -9,7 +9,39 @@ There are two ways to build/install this package.
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Clone this repo and cd into it::
+
+       git clone https://gitlab.is.tue.mpg.de/robotics/trifinger_simulation.git
+       cd trifinger_simulation
+
+2. Set up the conda env::
+
+       export PYTHONNOUSERSITE=True
```
Comment (luator) [inline docs/installation.rst]: I assume you removed the note on `PYTHONNOUSERSITE` below because it is not so important in a non-challenge context?  In this case, could it be removed here as well?

File: python/trifinger_simulation/sim_finger.py (hunk 4 above)
Comment (luator) [reply python/trifinger_simulation/sim_finger.py:30]: sure, this is not critical

File: docs/installation.rst (hunk 5 above)
Comment (shrutij01) [reply docs/installation.rst]: Okay

File: python/trifinger_simulation/finger_types_data.py [hunk 6]
```diff
@@ -32,6 +32,10 @@ def get_valid_finger_types():
     Returns:
         list: List of supported finger types.
     """
+    print("Among these listed finger types, 'single' and 'tri'"
+          " are deprecated. Use 'fingerone' and 'trifingerone'"
+          " instead respectively."
+            )
```
Comment (luator) [inline python/trifinger_simulation/finger_types_data.py]: I'm still strongly against this.  If it is a regular print now, it will be even more annoying.

I'm not sure where this is used (if it would break things) but how about we simply remove these entries from the list before returning it?

File: python/trifinger_simulation/gym_wrapper/envs/finger_push.py [hunk 7]
```diff
@@ -8,53 +8,72 @@
...
-        if finger_type == "single":
-            self.num_fingers = 1
-        else:
-            self.num_fingers = 3
-        simulation_rate_s = 0.004
-        self.steps_per_control = int(round(control_rate_s / simulation_rate_s))
+        self.num_fingers = finger_types_data.get_number_of_fingers(finger_type)
+
+        #: the number of times the same action is to be applied to
+        #: the robot.
```
Comment (luator) [inline python/trifinger_simulation/gym_wrapper/envs/finger_push.py:56]:
```suggestion
        #: the robot in one step.
```

File: python/trifinger_simulation/gym_wrapper/envs/finger_push.py [hunk 8]
```diff
@@ -63,23 +82,17 @@ def __init__(
...
+        # or action type, whereas all the values in the observation and action
+        # spaces lie between 1 and -1
         self.spaces = FingerSpaces(
             num_fingers=self.num_fingers,
             observations_keys=self.observations_keys,
             observations_sizes=self.observations_sizes,
             separate_goals=False,
         )
 
-        self.finger = SimFinger(
```
Comment (luator) [inline python/trifinger_simulation/gym_wrapper/envs/finger_push.py:73]: Is this removed intentionally? Maybe I'm missing it but it doesn't seem to be created anywhere now.

File: python/trifinger_simulation/gym_wrapper/envs/finger_push.py [hunk 9]
```diff
@@ -8,53 +8,72 @@
...
 
         Args:
-            control_rate_s (float): the rate at which the env step runs
-            finger-type (str "single"/"tri"): to train on the "single"
-                or the "tri" finger
+            control_rate_s (float): the rate at which step method of the env
+                will run. This is used to compute the number of times the same
+                action has to be applied to the robot in order to say that at
+                the end of this action loop, the robot would have reached the
+                desired action.
```
Comment (luator) [inline python/trifinger_simulation/gym_wrapper/envs/finger_push.py:39]: I find this a bit confusing.  I think the following would be a bit clearer from a technical point of few (just a suggestion, though).

```suggestion
            control_rate_s (float): the rate (in seconds) at which step method of the env
                will run. The actual robot controller may run at a higher rate,
                so this is used to compute the number of robot control updates
                per environment step.
```

File: python/trifinger_simulation/gym_wrapper/envs/finger_push.py (hunk 8 above)
Comment (shrutij01) [reply python/trifinger_simulation/gym_wrapper/envs/finger_push.py:73]: It is created in the first line of the __init__ method...

File: python/trifinger_simulation/finger_types_data.py (hunk 6 above)
Comment (shrutij01) [reply python/trifinger_simulation/finger_types_data.py]: I think this is more of a utitlity method that the user may call once before actually doing anything. And "single" and "tri" are still valid finger types. So, this message will only get printed once anyway.

File: python/trifinger_simulation/gym_wrapper/envs/finger_push.py (hunk 9 above)
Comment (shrutij01) [reply python/trifinger_simulation/gym_wrapper/envs/finger_push.py:39]: ok, i agree

File: python/trifinger_simulation/gym_wrapper/envs/finger_push.py (hunk 8 above)
Comment (luator) [reply python/trifinger_simulation/gym_wrapper/envs/finger_push.py:73]: Oh, yes... sorry, I somehow missed that.
